src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -101,8 +135,8 @@ public sealed class ActivitySource : IDisposable
         public string? Version { get { throw null; } }
         public bool HasListeners() { throw null; }
         public System.Diagnostics.Activity? StartActivity(string name, System.Diagnostics.ActivityKind kind = ActivityKind.Internal)  { throw null; }
-        public System.Diagnostics.Activity? StartActivity(string name, System.Diagnostics.ActivityKind kind, System.Diagnostics.ActivityContext parentContext, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null, System.DateTimeOffset startTime = default) { throw null; }
-        public System.Diagnostics.Activity? StartActivity(string name, System.Diagnostics.ActivityKind kind, string parentId, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null, System.DateTimeOffset startTime = default) { throw null; }
+        public System.Diagnostics.Activity? StartActivity(string name, System.Diagnostics.ActivityKind kind, System.Diagnostics.ActivityContext parentContext, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null, System.DateTimeOffset startTime = default) { throw null; }
+        public System.Diagnostics.Activity? StartActivity(string name, System.Diagnostics.ActivityKind kind, string parentId, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object?>>? tags = null, System.Collections.Generic.IEnumerable<System.Diagnostics.ActivityLink>? links = null, System.DateTimeOffset startTime = default) { throw null; }
```

Comment:
@tarekgh Hey I just noticed working on some code in OTel that you can't specify events in the StartActivity call. You can specify pretty much everything else. Should we add events too?

Reply:
Also there is no AddLink method. Links you have to specify at creation?

Reply:
> I just noticed working on some code in OTel that you can't specify events in the StartActivity call. You can specify pretty much everything else. Should we add events too?

StartActivity arguments mainly the things needed for sampling decision to decide if want to create the activity or just avoid that and sample it out. If you look at the OT sampling specs, you'll find events are not needed there https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#sampling. But anyway, you can add events after creating the activity. 

> Also there is no AddLink method. Links you have to specify at creation?

This was proposed before and then got removed before there was a claim that shouldn't allow adding links after creating and starting the activity. OT sepcs saying `During the Span creation user MUST have the ability to record links to other Spans. Linked Spans can be from the same or a different trace.`. so, the spec is explicit mentioning the links should be provided during the Span creation. But anyway, later if we find any scenario need to add links after starting the activity, we can consider it.

CC @cijothomas @reyang

Reply:
What Tarek said is my understanding as well.

Reply:
That all makes sense. What I'm doing right now is working on a unit test. It needs the Activity created very specifically. So it's not a normal use case. I think I can make it work with reflection. The links are a bit of a challenge. I'll mention you guys on the change once I have it working so you can see what I'm up against, and then we can decide if any changes are warranted.